nntrainer/models/neuralnet.h

```diff
@@ -254,7 +256,7 @@ class NeuralNetwork : public ml::train::Model {
    * @param[in] X input tensor
    * @retval shared_ptr<const Tensor>
    */
-  sharedConstTensors inference(sharedConstTensors X);
+  sharedConstTensors inference(sharedConstTensors X, bool free_mem = true);
```

Comment:
If we keep using inference which is initialized, then this free_mem should be false.. then it does make sure not to allocate the memory? ( may be using tensors_allocated and weight_allocated variable?)

Reply:
> then it does make sure not to allocate the memory?

Yes, if `free_mem` is passed as false, then no new memory is allocated (using exactly tensors_allocated and weight_allocated) resulting in reduction of used bandwidth as well as faster inference. 
`free_mem` can be set to true with last inference or that memory can be freed manually with `model.deallocate()`.